emr_launcher.py

```diff
@@ -103,7 +104,17 @@ def handler(event: dict = {}, context: object = None) -> dict:
     """Launches an EMR cluster with the provided configuration."""
     logger = configure_log()
 
+    sns_message = event["Records"][0]["Sns"]
+    payload = json.loads(sns_message["Message"])
+    correlation_id = payload["correlation_id"]
+
     cluster_config = read_config("cluster")
+    # Obtain Spark arguments for EMR from the config and add correlation id to it
```

Comment:
this block needs to be after `cluster_config.update(read_config("steps", False))` because at this point you wouldn't have Steps key